### PR TITLE
Thorium Command LIne refactor - concurrency and fixes opds/import redux-saga call

### LINKED ISSUE #2663: Error when running `opds` CLI command
Using Thorium 3.0.0 AppImage on Arch Linux, running this command:
```bash
./Thorium-3.0.0.AppImage opds Gallica http://gallica.bnf.fr/opds
```
I get the following error:

```
OPDS import ERROR: Error: No matching bindings found for serviceIdentifier: Symbol(saga-middlewaire)
```

IIUC it is comming from here:
https://github.com/edrlab/thorium-reader/blob/b5d1e740d995c1317caec3c31b17daba386e2bd9/src/main/diSymbolTable.ts#L4

and there is a potential typo there (`"saga-middlewaire"` where it should be `"saga-middleware"`).

### PR DESCRIPTION
**Pull request by panaC**
fixes #2663

### CODE REVIEW

**Comment by panaC** [comment]
@danielweck In this PR you will find : 
- got the lock with an early exit, when a process does not have the lock it exit with code 0 immediately, no computation/change is done on incoming thorium process, argv is send to got the lock process
- fixes opds and import commands, these 2 commands was not migrated from 2.0.0 version and no tested so far. persistState routine was not called and command was executed from incomming process that did not got the lock
- fixes opds and import commands concurrency, these commands can be chained foreground.
- createStore concurrency with a global process.on promise rejection handler

File: src/main/di.ts
Comment (panaC) [reply src/main/di.ts:219]:
```
diff --git a/src/main/di.ts b/src/main/di.ts
index a5f96940..2d540dfe 100644
--- a/src/main/di.ts
+++ b/src/main/di.ts
@@ -181,7 +181,7 @@ const container = new Container();
 
 
 const getStorePromiseFn = async () => {
-    createStoreProcessLock.lock();
+    // createStoreProcessLock.lock();
 
     debug("initStore");
     const [store, sagaMiddleware] = await initStore();
diff --git a/src/main/redux/store/memory.ts b/src/main/redux/store/memory.ts
index 817ab2bb..9be00b4c 100644
--- a/src/main/redux/store/memory.ts
+++ b/src/main/redux/store/memory.ts
@@ -92,11 +92,19 @@ const test = (stateRaw: any): stateRaw is PersistRootState => {
     return stateRaw;
 };
 
+
+let ___test = false;
+
 export async function initStore()
     : Promise<[Store<RootState>, SagaMiddleware<object>]> {
 
     let reduxState: PersistRootState | undefined;
 
+    if (!___test) {
+        ___test = true;
+        await new Promise((res) => setTimeout(() => res(undefined), 10*1000));
+    }
+
     debug("");
     debug("MEMORY INIT STORE");

```

```
npm run start:dev:main:electron -- opds Gallica "http://gallica.bnf.fr/opds" &
npm run start:dev:main:electron &
```

image

File: src/main/cli/index.ts
Comment (panaC) [reply src/main/cli/index.ts:100]: thanks

**Review by danielweck — APPROVED** [review]

File: src/main/di.ts
Comment (panaC) [reply src/main/di.ts]: you're right, global awaiting promise, do the lock

**Comment by danielweck** [comment]
Hello @panaC I approved the PR after the minor corrections, feel free to merge! :) (and thank you for figuring out the Electron app lifecycle between CLI vs. single-instance-lock app GUI. Let's hope this works on Windows and Linux too 👍

**Pull request merged**
